src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -120,46 +135,119 @@ public int BufferGrowLimit
         {
             get
             {
-                return growLimit;
+                return _bufferGrowLimit;
             }
 
             set
             {
-                growLimit = value;
+                _bufferGrowLimit = value;
                 GrowBufferAsNeeded = (value >= BufferSize);
             }
         }
 
         /// <summary>
-        /// Initializes the target by hooking up the NLogHttpModule BeginRequest and EndRequest events.
+        /// Context for DI
+        /// </summary>
+        private IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Provides access to the current request HttpContext.
         /// </summary>
-        protected override void InitializeTarget()
+        /// <returns>HttpContextAccessor or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        internal IHttpContextAccessor HttpContextAccessor
+        {
+            get => _httpContextAccessor ?? (_httpContextAccessor = RetrieveHttpContextAccessor(ResolveService<IServiceProvider>(), LoggingConfiguration));
+            set => _httpContextAccessor = value;
+        }
+
+        private IHttpContextAccessor RetrieveHttpContextAccessor(
+            IServiceProvider serviceProvider,
+            LoggingConfiguration loggingConfiguration)
         {
-            base.InitializeTarget();
+#if !ASP_NET_CORE
+            return new DefaultHttpContextAccessor();
+#else
+            return ServiceLocator.ResolveService<IHttpContextAccessor>(serviceProvider, loggingConfiguration);
```

Comment:
Can the input parameters be removed so it just becomes:
```c#
return ServiceLocator.ResolveService<IHttpContextAccessor>(ResolveService<IServiceProvider>(), LoggingConfiguration);
```

Reply:
Completed.  Also removed many #If ASP_NET_CORE statements and removed [NLogConfigurationIgnoreProperty] on HttpContextAccessor since this is internal and not public.

Reply:
Does this HttpContextAccessor improvement also need to be performed on AspNetLayoutRenderBase?

Reply:
No it is okay to leave `AspNetLayoutRenderBase` alone.